doc/sphinx/dg.rst

```diff
@@ -1,495 +1,137 @@
 
 The central source files of the build system are the following:
 
--  ``configure.ac``
-
--  ``config/*.m4``
-
--  ``Makefile.am``
-
--  the different ``Makefile.am`` in the subdirectories
-
-To change the behaviour of the build system, you have to modify any
-these files.
-
-.. _running_bootstrapsh:
-
-Running ``bootstrap.sh``
-^^^^^^^^^^^^^^^^^^^^^^^^
-
-The script ``bootstrap.sh`` in the top level source directory can be
-used to run ``automake``, ``autoconf`` and the associated tools to
-generate the ``configure`` script and the ``Makefile.in`` used during
-compilation. Once ``bootstrap.sh``, the |es| development code can be
-build and used like the release code.
-
-.. _creating_a_distribution_package:
-
-Creating a Distribution Package
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-As described in the User’s Guide, to create a ``.tar.gz`` distribution
-file that contains all required files, you can simply run ``make dist``.
-This will bundle all files and put them into an archive with the name
-``espresso-version.tar.gz``.
-
-Even better, you can also run ``make distcheck``. This will not only
-create the distribution file, but it will also thoroughly check the
-created distribution, *i.e.*\ it will try to
-
--  unpack the distro into a new directory
-
--  configure and build it (``configure``)
-
--  run the testsuite (``make check``)
-
--  install it into a new directory (``make install``)
+-  ``CMakeList.txt``
 
--  uninstall it (``make uninstall``)
+-  Contents of the ``cmake`` directory
 
-Whenever something goes wrong in these checks, it will give an error
-message that describes the problem. When everything goes fine, you can
-be relatively sure that you have a useful |es| distribution package.
+-  The CMakeList.txt files in the ``src/``, ``doc/``, and ``testsuite/`` directories and their sub-directories
 
-In some cases, it might be necessary to pass some options to the run of
-``configure`` done by ``make distcheck``>. To these ends, the
-environment variable ``DISTCHECK_CONFIGURE_FLAGS`` can be set to the
-required options.
+The most common reasons for editing these files are:
+-  Adding new source files
+-  Adding new external dependencies
 
-Example
-"""""""
-
-::
-
-    DISTCHECK_CONFIGURE_FLAGS="--without-mpi CPPFLAGS=\"-I /usr/include/tcl8.4\"" \
-      make distcheck
-
-.. _build_system_adding_new_files:
-
-Adding New Files
-^^^^^^^^^^^^^^^^
-
-To add new files to |es| (like C source files or header files) you
-need to do the following:
-
--  Add the files to the ``Makefile.am`` in the same directory
-
--  Run ``bootstrap.sh`` in the source directory
-
--  Check the distribution by using ``make distcheck``
+Adding New Source Files
+-----------------------
 
--  Add the files to the Git repository
+To add new files to |es| (like C++ source files or header files) you
+need to look at the CMakeList.txt in the directory where the file is located.
+* Please note that .hpp-header files usually do not have to be added to CMakeList.txt
```

Comment:
Remove "usually".  Header files *never* have to be added to CMakeLists.txt.  Only translation units have to be added.